sklearn/tests/test_metaestimators.py

```diff
@@ -44,6 +49,38 @@ def __init__(self, name, construct, skip_methods=(),
                                 'predict'])
 ]
 
+TESTED_META = [
+    # pipeline
+    # this fails because tuple is converted to list in fit:
+    Pipeline((('ss', StandardScaler()),)),
+    # all pipelines fail because they don't clone:
+    Pipeline([('ss', StandardScaler())]),
+    make_pipeline(StandardScaler(), LogisticRegression()),
+    # union
+    make_union(StandardScaler()),
+    # union and pipeline
+    make_pipeline(make_union(PCA(), StandardScaler()),
+                  LogisticRegression()),
+    # pipeline with clustering
+    make_pipeline(KMeans(random_state=0)),
+    # SelectFromModel
+    make_pipeline(SelectFromModel(LogisticRegression(), threshold=-np.inf),
+                  LogisticRegression()),
+    # grid-search
+    GridSearchCV(LogisticRegression(), {'C': [0.1, 1]}, cv=2),
+    # will fail tragically
+    # make_pipeline(StandardScaler(), None)
+]
+
+
+@parametrize_with_checks(TESTED_META)
+def test_metaestimators_check_estimator(estimator, check):
+    if check.func.__name__ in ["check_estimators_overwrite_params",
```

Comment:
This only applies to pipeline and feature union, so why are we testing everything together here?

Reply:
Would you make a separate test for each meta-estimator? Or one for pipeline and feature union and one for all the rest?

Reply:
I don't see why you wouldn't put each in their test file

Reply:
add `and isinstance(estimator, (...))`?